src/Queue.js

```diff
@@ -19,56 +19,113 @@ class Queue {
 
     // Enqueues the given process. Return the enqueue'd process
     enqueue(process) {
-
+        process.queue = this;
+        return this.processes.push(process);
     }
 
     // Dequeues the next process in the queue. Return the dequeue'd process
     dequeue() {
-
+        return this.processes.shift();
     }
 
     // Return the least-recently added process without removing it from the list of processes
     peek() {
-
+        return this.processes[0];
     }
 
     isEmpty() {
-
+        return this.processes.length < 1 || this.processes === undefined
+            ? true
+            : false;
     }
 
     getPriorityLevel() {
-
+        return this.priorityLevel;
     }
 
     getQueueType() {
-
+        return this.queueType;
     }
 
     // Manages a process's execution for the given amount of time
     // Processes that have had their states changed should not be affected
-    // Once a process has received the alloted time, it needs to be dequeue'd and 
+    // Once a process has received the alloted time, it needs to be dequeue'd and
     // then handled accordingly, depending on whether it has finished executing or not
     manageTimeSlice(currentProcess, time) {
-
+        // if (currentProcess.isStateChanged()){
+        //     this.quantumClock = 0;
+        //     return;
+        // }
+
+        // this.quantumClock += time;
+        // if(this.quantumClock >= this.quantum){
+        //     this.quantumClock = 0;
+        //     this.dequeue();
+
+        //     if(!currentProcess.isFinished()){
+        //         this.scheduler.handleInterrupt(this, currentProcess, SchedulerInterrupt.PROCESS_BLOCKED);
+        //     } else {
+        //         console.log("process complete");
+        //     }
+        // }
+        if (!currentProcess.stateChanged) {
```

Comment:
Hopefully out discussion helped clarify things. Your way of thinking is reversed here